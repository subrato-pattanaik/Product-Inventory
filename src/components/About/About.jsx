import React from "react";
import { Carousel, Container } from "react-bootstrap";
import EditProduct from "../../images/EditProduct.jpg";
import NewProduct from "../../images/NewProduct.jpg";
import PI from "../../images/PI.jpg";
import Viewproduct from "../../images/Viewproduct.jpg";
function About() {
  return (
    <Container fluid className="p-0" style={{ Height: "100vh !important" }}>
      <Carousel>
        <Carousel.Item>
          <img className="d-block w-100" src={PI} alt="First slide" />
        </Carousel.Item>
        <Carousel.Item>
          <img className="d-block w-100" src={NewProduct} alt="Second slide" />
        </Carousel.Item>
        <Carousel.Item>
          <img className="d-block w-100" src={EditProduct} alt="Third slide" />
        </Carousel.Item>

        <Carousel.Item>
          <img
            className="d-block w-100"
            src={Viewproduct}
            alt="TFourth slide"
          />
        </Carousel.Item>
      </Carousel>
    </Container>
  );
}

export default About;
