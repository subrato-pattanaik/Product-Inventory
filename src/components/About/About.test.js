import React from "react";
import { shallow } from "enzyme";

import About from "./About";

describe("About Page Testing", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<About />);
  });

  it("should test snapshot of About page", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should test image built-in component", () => {
    expect(wrapper.find("img").exists()).toEqual(true);
  });

  it("should test image using property selector built-in component", () => {
    expect(wrapper.find('[alt="First slide"]').exists()).toEqual(true);
  });

  it("should test image built-in component count", () => {
    expect(wrapper.find("img")).toHaveLength(4);
  });
});
