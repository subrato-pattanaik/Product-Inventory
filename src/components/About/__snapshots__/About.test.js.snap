// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Page Testing should test snapshot of About page 1`] = `
<Container
  className="p-0"
  fluid={true}
  style={
    Object {
      "Height": "100vh !important",
    }
  }
>
  <Carousel
    controls={true}
    defaultActiveIndex={0}
    fade={false}
    indicators={true}
    interval={5000}
    keyboard={true}
    nextIcon={
      <span
        aria-hidden="true"
        className="carousel-control-next-icon"
      />
    }
    nextLabel="Next"
    pause="hover"
    prevIcon={
      <span
        aria-hidden="true"
        className="carousel-control-prev-icon"
      />
    }
    prevLabel="Previous"
    slide={true}
    touch={true}
    wrap={true}
  >
    <CarouselItem>
      <img
        alt="First slide"
        className="d-block w-100"
        src="PI.jpg"
      />
    </CarouselItem>
    <CarouselItem>
      <img
        alt="Second slide"
        className="d-block w-100"
        src="NewProduct.jpg"
      />
    </CarouselItem>
    <CarouselItem>
      <img
        alt="Third slide"
        className="d-block w-100"
        src="EditProduct.jpg"
      />
    </CarouselItem>
    <CarouselItem>
      <img
        alt="TFourth slide"
        className="d-block w-100"
        src="Viewproduct.jpg"
      />
    </CarouselItem>
  </Carousel>
</Container>
`;
